migrate/starter/starter.go

```diff
@@ -61,6 +62,11 @@ func checkStartVersion(args []string) version {
 	if err != nil {
 		return internalV2
 	}
+	if fs.Lookup("version").Value.String() == "true" {
+		fmt.Println("etcd version:", etcdversion.Version, "(allow legacy mode)")
```

Comment:
The `--version` flag should indicate the version of the binary.

>I have concerns that users have false thoughts that etcd v2.0 can run v0.4 data dir directly.

Is that not true? I thought the point of etcd v2.0 was that it could run both (by exec'ing the older etcd).